### ISSUE TITLE
Use XEM to help determine season packs for Anime.

### ISSUE THREAD

**Issue by Ravencentric** [Sonarr/Sonarr#5195]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe

Currently, it's nearly impossible to grab season packs from Anime trackers like Nyaa/AB due to the common way of using season titles rather than a season number to distinguish between seasons. I've described this in more detail along with examples and stats here - https://github.com/Prowlarr/Prowlarr/issues/1052. It's a bit better on AB because of https://github.com/Prowlarr/Prowlarr/pull/329 but Nyaa lacks this entirely.

### Describe the solution you'd like

Using XEM along with https://github.com/Prowlarr/Prowlarr/pull/329 (on both Nyaa and AB).
Currently, AB works like this:
```
Dr. Stone -> Dr. Stone S01 #Correct Guess, leveraging https://github.com/Prowlarr/Prowlarr/pull/329
Dr. Stone: Stone Wars -> Dr. Stone: Stone Wars S01 #Incorrect Guess, leveraging https://github.com/Prowlarr/Prowlarr/pull/329
```
But XEM already knows `Dr. Stone: Stone Wars` is the `S02` of `Dr. Stone`:
![image](https://user-images.githubusercontent.com/78981416/194944109-80c2dd8c-1506-4bf8-8876-8470c4e121b0.png)

So, is it *possible* to use XEM's data to determine the season? 
Ideal results for both AB and Nyaa:
```
Dr. Stone -> Dr. Stone S01             #Correct Guess, leveraging https://github.com/Prowlarr/Prowlarr/pull/329
Dr. Stone: Stone Wars -> Dr. Stone S02 #Correct Guess, leveraging https://github.com/Prowlarr/Prowlarr/pull/329 and XEM
```

### Describe alternatives you've considered

Manually downloading and importing.

### Anything else?

Using releases on [Seadex](https://releases.moe) (an index of best releases for each anime) and Nyaa.si and processing them to see whether they fail, match or parse to get an idea of the above issue.

Out of 1374 Seadex releases on Nyaa

[121 matched (8.8%)](https://gist.github.com/Ravencentric/16f153efe0788d9811ea420cafe9bfc6)
[133 parsed wrong (9.7%)](https://gist.github.com/Ravencentric/b8529289c4f84b2090df8a20f357c320)
[1120 failed to parse (81.5%) ](https://gist.github.com/Ravencentric/d030368e5da4ee281973dbb276607c96)

Note: Seadex has no distinction between Shows and Movies aside from their titles so the `failed to parse` and `parsed wrong` results, unfortunately, contain about a maximum of 200 movie releases combined. Even after accounting for that, the numbers are pretty bad:

121 matched (10.3%)
119 parsed wrong (10.1%)
934 failed to parse (79.5%))

Here's the result of a little experiment where `S01` was appended to all the Nyaa titles.

[718 matched (52.2%)](https://gist.github.com/Ravencentric/c274722c68b19846b951fe9a38bf0c39)
[519 parsed but not matched (43.4%)](https://gist.github.com/Ravencentric/585b40d86ca78b60d54e51d6cf74c57c)
[60 failed to parse (4.3%)](https://gist.github.com/Ravencentric/fe56cf125ee580d490e0d53776afc647)

This was just an inflexible "fix" but it goes to show that if this issue can be fixed, the success rate will be several times more.

**Comment by markus101** [comment]
For parsing this would require a significant amount of work from the parsing down to mapping those results to series/episodes, for searching it would probably be easier, but the results wouldn't be parseable, though it's not clear exactly what aspect this request is for.

Unfortunately none of your links work so we can't see what releases you're talking about.

**Comment by Ravencentric** [comment]
Sorry, updated the links.
What I'm trying to propose is something along the lines of the followng:

Assuming two releases of [Dr Stone](https://thexem.info/xem/show/5041):
- `[Group] Dr. Stone (BD 1080p HEVC x265 10-bit FLAC) [Dual-Audio]`
- `[Group] Dr. Stone: Stone Wars (BD 1080p HEVC x265 10-bit FLAC) [Dual-Audio]`

### Current Behavior:

**NYAA**:

Both fail as Sonarr is unable to detect them as season packs.

**ANIMEBYTES**:

Prowlarr tries and guesses the season pack with [Sonarr Compatibility Enabled](https://i.imgur.com/xdm15Lc.png) (https://github.com/Prowlarr/Prowlarr/pull/329).
- `[Group] Dr. STONE S01 [Blu-ray][MKV][h265 10-bit][1080p][FLAC 2.0][Dual Audio][Softsubs (Group)]` - A succesful match.
- `[Group] Dr. STONE: STONE WARS S01 [Blu-ray][MKV][h265 10-bit][1080p][FLAC 2.0][Dual Audio][Softsubs (Group)]` - Fails since `Dr. STONE: STONE WARS` is Season 2 of `Dr. Stone`

**However, XEM has the alias `Dr. STONE: STONE WARS` as Season 2. So assuming that can be pulled here, It can be used to help match `Dr. STONE: STONE WARS` correctly to Season 2.**

Note: [Sonarr Compatibility Mode](https://i.imgur.com/xdm15Lc.png) is non existent in Nyaa and only exists for AB which means before this could be done, Nyaa also needs said compatibility mode.

### Proposed Solution:
- Replicate the AB's implementation to Nyaa (https://github.com/Prowlarr/Prowlarr/issues/1052), then add the XEM improvements to both.

### Why?
Anime does not use the standard `S0X` to define a season pack. They use *titles*. `Dr Stone` is S01, while `Dr Stone: Stone Wars` is S02. There's only a minority of Groups that are defining season packs like `Season X` or  `S0X`.

**Comment by Enlinks** [comment]
I would like to add, being able to leverage XEM additionally for individual episodes would also be greatly beneficial. As it currently stands I had to stop monitoring Season 1 and 2 of Bleach due to RSS for Season 17. I was getting some matches that are marked as S01E0x and/or S02E0x and would overwrite my existing Season 1 and 2 files as the new Season is in 4k vs Season 1 and 2.

However if the XEM data was leveraged it either would have failed to match (would have preferred that is overwriting S01 and 2) or correctly matched itself to S17 instead. [Bleach seasion mapping](https://thexem.info/xem/show/5476)

Example match I got on S01 vs matching S17:
"Bleach - Thousand-Year Blood War - S01E01 - 2160p WEB H.264 -NanDesuKa (B-Global)"
